R/bayesfactor_inclusion.R

```diff
@@ -0,0 +1,204 @@
+
+    # models without effect
+    mwithoutprior <- sum(df.model_temp[-mwith,'priorProbs'])
+    mwithoutpost <- sum(df.model_temp[-mwith,'postProbs'])
+
+    # Save results
+    df.effect$Pinc[effnames==eff]  <- mwithprior
+    df.effect$PincD[effnames==eff] <- mwithpost
+    df.effect$BFinc[effnames==eff] <- (mwithpost/mwithoutpost)/(mwithprior/mwithoutprior)
+  }
+
+  df.effect$BFinc <- log(df.effect$BFinc)
+  df.effect <- df.effect[,-1,drop = FALSE]
+  colnames(df.effect) <- c("P.Inc.prior","P.Inc.posterior","log.BF.Inc")
+  rownames(df.effect) <- effnames
+
+
+  class(df.effect) <- c('BFinc',class(df.effect))
+  attr(df.effect,'matched') <- match_models
+  attr(df.effect,'priorOdds') <- prior_odds
+
+  return(df.effect)
+}
+
+bayesfactor_inclusion.BFBayesFactor <- function(models, match_models = FALSE, prior_odds = NULL) {
+  models <- bayesfactor_models.BFBayesFactor(models)
+  bayesfactor_inclusion.BFGrid(models, match_models = match_models, prior_odds = prior_odds)
+}
+
+#' @importFrom stats as.formula terms setNames
+get_model_table <- function(BFGrid, priorOdds = NULL){
+  denominator <- attr(BFGrid,'denominator')
+  BFGrid <- rbind(BFGrid[denominator,],BFGrid[-denominator,])
+  attr(BFGrid,'denominator') <- 1
+
+  # Prior and post odds
+  Modelnames <- BFGrid$Model
+  if (!is.null(priorOdds)) {
+    priorOdds <- c(1, priorOdds)
+  } else {
+    priorOdds <- rep(1, length(Modelnames))
+  }
+
+  posterior_odds <- priorOdds * exp(BFGrid$log.BF)
+
+  priorProbs <- priorOdds / sum(priorOdds)
+  postProbs <- posterior_odds / sum(posterior_odds)
+
+  df.model <- data.frame(Modelnames,
+                         priorProbs,
+                         postProbs,
+                         stringsAsFactors = FALSE)
+
+  # add effects table
+  make_terms <- function(formula) {
+    formula.f <- stats::as.formula(paste0('~', formula))
+    all.terms <- attr(stats::terms(formula.f), "term.labels")
+
+    fix_trms <- all.terms[!grepl("\\|", all.terms)] # no random
+
+    random_parts <- paste0(all.terms[grepl("\\|", all.terms)]) # only random
+    if (length(random_parts) == 0) {
+      return(stats::setNames(fix_trms, fix_trms))
+    }
+
+    random_units <- sub("^.+\\|\\s+", "", random_parts)
+    tmp_random <- lapply(
+      sub("\\|.+$", "", random_parts),
+      function(x) stats::as.formula(paste0("~", x))
+    )
+
+    rand_print <- rand_trms <- vector("list", length(random_parts))
+
+    for (i in seq_along(random_parts)) {
+      tmp_trms <- attr(terms.formula(tmp_random[[i]]), "term.labels")
+
+      if (!any(unlist(strsplit(as.character(tmp_random[[i]])[[2]], ' \\+ ')) == "0"))
+        tmp_trms <- c("1", tmp_trms)
+
+      rand_trms[[i]] <- paste0(tmp_trms, ':', random_units[[i]])
+      rand_print[[i]] <- paste0("(", tmp_trms, "|", random_units[[i]], ")")
+    }
+
+    stats::setNames(
+      c(fix_trms,unlist(rand_trms)),
+      c(fix_trms,unlist(rand_print))
+    )
+  }
+
+  print_terms <- character()
```

Comment:
`print_terms` is defined but not used...

Reply:
removed